plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/TrinoFileSystemCacheStats.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.fs;
+
+import io.airlift.stats.CounterStat;
+import org.weakref.jmx.Managed;
+import org.weakref.jmx.Nested;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+public class TrinoFileSystemCacheStats
+{
+    private final CounterStat getCalls = new CounterStat();
+    private final CounterStat getUniqueCalls = new CounterStat();
+    private final CounterStat getCallsFailed = new CounterStat();
+    private final CounterStat removeCalls = new CounterStat();
+    private final AtomicLong cacheSize = new AtomicLong();
```

Comment:
maybe remove this field and replace it with pull-based approach?

Reply:
I wanted to confirm I have the right idea before implementing. 

With a pull-based approach, we would no longer have calls such as `stats.setCacheSize` but instead  `TrinoFileSystemCacheStats#getCacheSize` would just call `TrinoFileSystemCache#getCacheSize`?

Reply:
pushed fixup with the change